api/src/OpenTelemetry/Trace/Core.hs

```diff
@@ -278,8 +279,8 @@ createSpanWithoutCallStack t ctxt n args@SpanArguments {..} = liftIO $ do
                           emptyAttributes
                           (H.unions [additionalInfo, attrs, attributes])
                     , spanLinks =
-                        let limitedLinks = fromMaybe 128 (linkCountLimit $ tracerProviderSpanLimits $ tracerProvider t)
-                        in frozenBoundedCollection limitedLinks $ fmap freezeLink links
+                        let emptyLinks = emptyAppendOnlyBoundedCollection $ fromMaybe 128 (linkCountLimit $ tracerProviderSpanLimits $ tracerProvider t)
+                        in foldl (\c l -> appendToBoundedCollection c l) emptyLinks (fmap (freezeLink t) links)
```

Comment:
Just want to make sure I understand: before, the collection itself and every link inside was frozen. Now the links inside are still each frozen, but the collection itself can be appended to. It makes sense you'd want to do this, but what are the ramifications, if any, of the collection no longer being frozen? Did it not need to be frozen for any particular reason originally?

Reply:
This is due to the fact that v0.7.0 of the spec specified this behaviour:

> During the Span creation user MUST have the ability to record links to other Spans. Linked Spans can be from the same or a different trace. See [Links description](https://github.com/open-telemetry/opentelemetry-specification/blob/v0.7.0/specification/overview.md#links-between-spans). Links cannot be added after Span creation.

Reply:
Aha, that makes a ton of sense. I had wondered why it was like that.